rotkehlchen/chain/evm/decoding/uniswap/utils.py

```diff
@@ -101,3 +110,53 @@ def decode_basic_uniswap_info(
         events_list=decoded_events,
     )
     return DEFAULT_DECODING_OUTPUT
+
+
+def get_uniswap_v2_position_price(
+        pool_token: 'EvmToken',
+        price: 'PoolPrice',
+        pool_token_amount: FVal,
+        evm_inquirer: 'EvmNodeInquirer',
+        block_identifier: BlockIdentifier = 'latest',
+) -> Price:
+    try:
+        total_supply = evm_inquirer.call_contract(
+            contract_address=pool_token.evm_address,
+            abi=evm_inquirer.contracts.abi('UNISWAP_V2_LP'),
+            method_name='totalSupply',
+            block_identifier=block_identifier,
+            call_order=evm_inquirer.archive_nodes_call_order(),
+        )
+
+        reserve_0, _, _ = evm_inquirer.call_contract(
+            contract_address=pool_token.evm_address,
+            abi=evm_inquirer.contracts.abi('UNISWAP_V2_LP'),
+            method_name='getReserves',
+            block_identifier=block_identifier,
+            call_order=evm_inquirer.archive_nodes_call_order(),
+        )
```

Comment:
to obtain `PoolPrice` we already query the reserves. You can add that metadata to the struct and use it here. It is less things that you need to calculate